src/main/java/io/cryostat/recordings/RecordingArchiveHelper.java

```diff
@@ -220,6 +222,9 @@ public Future<ArchivedRecordingInfo> deleteRecording(String recordingName) {
                     .message(Map.of("recording", archivedRecordingInfo))
                     .build()
                     .send();
+            if (pathIsEmpty(parentPath)) {
+                FileUtils.deleteDirectory(parentPath.toFile());
```

Comment:
Better to use our `FileSystem` abstraction:

https://github.com/cryostatio/cryostat-core/blob/main/src/main/java/io/cryostat/core/sys/FileSystem.java

There's already an instance of this within this class for you to use.

This is preferred because it's easily mockable in unit tests, basically. I think this class provides methods that will let you delete the directory as well as a simple way to check if it is empty.